src/components/Nav/RouterComponent.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import { Redirect, Route, Switch } from 'react-router';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { Welcome } from '../../pages';
+import Nav from './Nav';
+
+const RouterComponent = (props) => {
+  const { token, list, loading, error, updateToken } = props;
+  return (
+    <Router>
+      <Switch>
+        <Route
+          exact
+          path="/:page"
+          render={(props) => (
+            <Nav
+              {...props}
+              token={token}
+              list={list}
+              loading={loading}
+              error={error}
+              userToken={token}
+              updateToken={updateToken}
+            />
+          )}
+        />
```

Comment:
Hey so I had a couple of comments about this. I would recommend against setting the path to be a variable like this, it will allow any url to work regardless or whether or not it is actually a page. I would suggest something like this instead:

```
          path={['/list', '/additems', '/shareyourtoken']}
```

Along with that, there needs to check to see whether or not a token is set, otherwise you can access these pages by going directly to them without a token:

```suggestion
        {props.token && (
          <Route
            exact
            path={['/list', '/additems', '/shareyourtoken']}
            render={(props) => (
              <Nav
                {...props}
                token={token}
                list={list}
                loading={loading}
                error={error}
                userToken={token}
                updateToken={updateToken}
              />
            )}
          />
        )}
```